### PR TITLE
Use "discard" when deleting read resources

### LINKED ISSUE #2015: Saying `delete` for a read resource during destroy is scary
I have a program which does `gcp.container.Cluster.get` to retrieve an existing cluster so that I can deploy new containers into it without having to provide a full `kubeconfig`.

When I go to `destroy` this stack - I see the following:

```
$ pulumi destroy
Previewing destroy of stack 'gke-20181002-dev'
Previewing changes:

     Type                            Name                           Plan       Info
 -   pulumi:pulumi:Stack             gke-20181002-gke-20181002-dev  delete
 -   ├─ kubernetes:core:Pod          nginx                          delete
 -   ├─ pulumi:providers:kubernetes  luke-cluster                   delete
 -   ├─ gcp:container:Cluster        luke-cluster                   delete
 -   └─ gcp:storage:Bucket           my-bucket                      delete

info: 5 changes previewed:
    - 5 resources to delete

Do you want to perform this destroy?
  yes
> no
  details
```

This is scary and a bit misleading.  It implies that we will delete the GKE Cluster.  But we will (I hope!) not actually delete it, just remove it from the checkpoint file.

Feels like we need to explain this more clearly.  In the context of a refresh, it is sensible to use `delete` to mean "remove from checkpoint".  But in the context of an update (or preview of an update), `delete` means something else.  We could consider `remove`, `forget`,etc.?  This is morally the opposite of `read` which shows up at the other end of this operation.

### PR DESCRIPTION
**Pull request by pgavlin**
In general, a "delete" in Pulumi is destroying an actual physical
resource. In the case of a read resource, however, the delete is
merely removing the resource from the stack; the physical resource
is not affected. These changes attempt to clarify this situation by
using the term "discard" rather than "delete".

Fixes #2015.

### CODE REVIEW

**Comment by joeduffy** [comment]
Remove is still only subtly different from delete. What would you think about "forget"?

**Comment by pgavlin** [comment]
What about "drop" or "discard"?

File: pkg/engine/lifecycle_test.go [hunk 1]
```diff
@@ -134,7 +134,8 @@ func (j *Journal) Snap(base *deploy.Snapshot) *deploy.Snapshot {
 
 		if e.Kind != JournalEntryOutputs {
 			switch e.Step.Op() {
-			case deploy.OpCreate, deploy.OpCreateReplacement, deploy.OpRead, deploy.OpReadReplacement, deploy.OpUpdate:
+			// nolint: lll
+			case deploy.OpCreate, deploy.OpCreateReplacement, deploy.OpRead, deploy.OpReadReplacement, deploy.OpUpdate, deploy.OpReadRemove:
```
[the remark this answers is not in the document]
Comment (pgavlin) [reply pkg/engine/lifecycle_test.go]: Yep--wrong line. Good catch.

File: pkg/resource/deploy/step.go [hunk 2]
```diff
@@ -247,6 +247,8 @@ func NewDeleteReplacementStep(plan *Plan, old *resource.State, pendingDelete boo
 func (s *DeleteStep) Op() StepOp {
 	if s.replacing {
 		return OpDeleteReplaced
```
[the remark this answers is not in the document]
Comment (pgavlin) [reply pkg/resource/deploy/step.go]: I believe that will show as `OpDeleteReplaced`. I'll have to think about whether or not we need an `OpRemoveReplaced`.

**Comment by pgavlin** [comment]
Ping.

**Pull request merged**